tests/functional/api/groups/members_test.py

```diff
@@ -62,179 +62,135 @@ def test_it_returns_empty_list_if_no_members_in_group(self, app):
 
-        app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=third_party_group.pubid, userid=third_party_user.userid
-            ),
-            headers=self.authorization_header(auth_client),
-        )
+        do_request(pubid=group.pubid, status=404)
 
-        res = app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=third_party_group.pubid, userid=third_party_user.userid
-            ),
-            headers=self.authorization_header(auth_client),
-        )
+    def test_it_errors_if_the_users_authority_doesnt_match(self, factories, do_request):
+        user = factories.User(authority="other")
 
-        assert third_party_user in third_party_group.members
-        assert res.status_code == 204
+        do_request(userid=user.userid, status=404)
 
-    def test_it_returns_404_if_authority_mismatch_on_user(
-        self, app, factories, auth_client
+    def test_it_errors_if_the_groups_and_users_authorities_both_dont_match(
+        self, factories, do_request
     ):
-        group = factories.Group()
-        user = factories.User(authority="somewhere-else.org")
+        # The user and group have the same authority but it is different from
+        # the authclient's authority.
+        group = factories.Group(authority="other")
+        user = factories.User(authority=group.authority)
 
-        res = app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=group.pubid, userid=user.userid
-            ),
-            headers=self.authorization_header(auth_client),
-            expect_errors=True,
-        )
+        do_request(pubid=group.pubid, userid=user.userid, status=404)
 
-        assert res.status_code == 404
-
-    def test_it_returns_404_if_malformed_userid(
-        self, app, factories, auth_client, db_session
+    def test_it_ignores_forward_users(
+        self, factories, authclient, headers, do_request, user, group
     ):
-        group = factories.Group()
-        db_session.commit()
+        forwarded_user = factories.User(authority=authclient.authority)
+        headers["X-Forwarded-User"] = forwarded_user.userid
 
-        res = app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=group.pubid, userid="foo@bar.com"
-            ),
-            headers=self.authorization_header(auth_client),
-            expect_errors=True,
-        )
+        do_request()
 
-        assert res.status_code == 404
+        # It has added the user from the URL to the group.
+        assert user in group.members
+        # It has *not* added the user from the X-Forwarded-User header to the group.
+        assert forwarded_user not in group.members
 
-    def test_it_returns_404_if_authority_mismatch_on_group(
-        self, app, factories, user, auth_client, db_session
+    @pytest.mark.xfail
+    def test_me_alias_with_forwarded_user(
```

Comment:
A future PR is going to incidentally add support for the "me" alias to the add-member API at which time this test will pass. For now I added a failing test